### PR TITLE
Remove openvpn

### PR DESCRIPTION
**Pull request by sjenning**
@csrwng @relyt0925 

Last step in move to konnectivity

### CODE REVIEW

File: cmd/install/assets/hypershift-operator/hypershift.openshift.io_hostedclusters.yaml [hunk 1]
```diff
@@ -258,7 +258,6 @@ spec:
                       description: Service identifies the type of service being published
                       enum:
                       - APIServer
-                      - VPN
```
Comment (ironcladlou) [inline cmd/install/assets/hypershift-operator/hypershift.openshift.io_hostedclusters.yaml:261]: Note: backwards incompatible API change

**Comment by relyt0925** [comment]
SUPER EXCITING :)

**Comment by relyt0925** [comment]
This looks good as is when http 2 health checking was enabled which fixes a deadlock condition in HTTP2 connections in golang (only way to recover is to restart component in older versions of go).

**Comment by relyt0925** [comment]
Upstream prs:
https://github.com/kubernetes-sigs/apiserver-network-proxy/pull/246
https://github.com/kubernetes-sigs/apiserver-network-proxy/pull/248

**Comment by sjenning** [comment]
Failed at CI infra setup before any hypershift stuff.  Will try again at a less busy time.

**Comment by sjenning** [comment]
It passes :tada:

**Pull request merged**